docs/design/client-configuration.md

```diff
@@ -0,0 +1,336 @@
+# Client Configuration
+
+* **Type**: Convention
+* **Author(s)**: Aaron Todd
+
+# Abstract
+
+This document describes the structure and conventions used for service client configuration. 
+
+# Conventions
+
+This section describes general conventions and guidelines to be used for service client configuration. See the section
+below for concrete example of these conventions in practice.
+
+* All generated service clients inherit from `SdkClientConfig.Builder`
+* Domain specific configuration (e.g. HTTP, tracing, etc) should be used as a "mixin" rather than form a hierarchy
+    * e.g. `HttpClientConfig` should not inherit from `SdkClientConfig`
+* Configuration interfaces should live in a subpackage `config` of whatever root package they belong to.
```

Comment:
Ya I'm on fence on this. The `runtime-core` package had thing split between a `client` subpackage and `config`. Both of which had "client" only related things in them. I opted to move them all to the `client` subpackage as it's more fitting. The thought or concern I have is an eventual server side component depending on `runtime-core`. It made sense in my head to move the "client" concerns to a "client" subpackage. Alas this creates the inconsistency in question. 

Rather than guess or go through multiple rounds, maybe we can just discuss what makes sense.